py/selenium/webdriver/remote/webelement.py

```diff
@@ -535,11 +540,13 @@ def send_keys(self, *value) -> None:
                 remote_files = []
                 for file in local_files:
                     remote_files.append(self._upload(file))
-                value = '\n'.join(remote_files)
+                output = '\n'.join(remote_files)
+        else:
+            output = value
 
         self._execute(Command.SEND_KEYS_TO_ELEMENT,
-                      {'text': "".join(keys_to_typing(value)),
-                       'value': keys_to_typing(value)})
+                      {'text': "".join(keys_to_typing(output)),
```

Comment:
This is causing `test-remote` execution to fail: https://github.com/SeleniumHQ/selenium/runs/4708398183

It looks like there's an `UnboundLocalError` issue because `output` isn't defined for condition: `None in local_files`

Reply:
i think i've fixed this in 41b47ead89f29aa5c9535d46385a94228f6d517e, but i'm a bit confused about this method in general; it seems like this function will perform _dramatically_ different things based on whether one entry in a list of filenames is not a local file, and now that i know this machinery is here, i'm worried about accidentally matching file names from my current working directory and accidentally sending their contents to my test runners

Reply:
So, this has to do with file uploads. If the element is a file type input element and the string passed in send keys is a path to a file, the driver will handle it as an upload.

Look at Point 8 here for remote end steps for element send keys: 
https://w3c.github.io/webdriver/#:~:text=the%20remote%20end%20steps%20for%20element%20send%20keys%20are

This gets weird when using a remote driver and the file is on a different system than the driver, which requires using a local file detector and uploading the file to the remote machine

Reply:
I haven't dug into the Python implementation. I'm wondering if it might make sense to have new methods in the bindings specifically for "upload" so we aren't overloading "send keys," but we're also looking at making a separate upload endpoint in w3c, so maybe that should be done first in case the user-facing implementation in the bindings is going to be different.

Reply:
i still don't think there's any action left to be taken here as part of this MR